homeassistant/components/ecovacs/vacuum.py

```diff
@@ -56,14 +64,18 @@ def __init__(self, device: sucks.VacBot) -> None:
         """Initialize the Ecovacs Vacuum."""
         self.device = device
         self.device.connect_and_wait_until_ready()
-        if self.device.vacuum.get("nick") is not None:
-            self._attr_name = str(self.device.vacuum["nick"])
+        vacuum = self.device.vacuum
+
+        self.error = None
+        self._attr_unique_id = vacuum.get("did")
+
+        if vacuum.get("nick") is not None:
```

Comment:
Like this, or maybe a walrus
```suggestion
        if "nick" in vacuum:
```

Reply:
would this be ok?
```python
self._attr_name = vacuum.get("nick", vacuum["did"])
```